coderd/httpapi/httpapi.go

```diff
@@ -132,3 +135,81 @@ func WebsocketCloseSprintf(format string, vars ...any) string {
 
 	return msg
 }
+
+func SetupSSE(rw http.ResponseWriter, r *http.Request) error {
+	h := rw.Header()
+	h.Set("Content-Type", "text/event-stream")
+	h.Set("Cache-Control", "no-cache")
+	h.Set("Connection", "keep-alive")
+	h.Set("X-Accel-Buffering", "no")
+
+	f, ok := rw.(http.Flusher)
+	if !ok {
+		return xerrors.New("http.ResponseWriter is not http.Flusher")
+	}
+
+	_, err := io.WriteString(rw, ": ping\n\n")
+	if err != nil {
+		return err
+	}
+	f.Flush()
```

Comment:
you could make this an anonymous function to avoid the duplication here and on line 167